Adds items on new/update business

istanfield09 opened the pull request:
Adds the ability to specify which items are
accepted by businesses while creating and
updating them.

istanfield09 commented on src/routes/AdminRoutes.php:
```diff
@@ -710,13 +711,24 @@ function fetch_val($key, $array, $default = 'undefined') {
 
         /* Update Mobile Database */
-        reuse_generateXML();
+        // reuse_generateXML();
```
This should be uncommented before merging; was causing 500 problems in my API because of missing configurations.

istanfield09 commented on public_html/js/BusFunct.js:
```diff
@@ -72,6 +104,8 @@ function addNewBusiness(){
   var phone = (document.getElementById("phone").value).replace(/\D/g,'');
   var website = document.getElementById("website").value;
+  var items = getSelectValues(document.getElementById("select-items"));
+  console.log(items);
```
Remove this.

istanfield09 commented on src/routes/AdminRoutes.php:
```diff
@@ -59,6 +59,35 @@ function fetch_val($key, $array, $default = 'undefined') {
+     *
+     * @apiName ReUseApp
+     * @apiGroup RUapi
+     
```
Fix this.

istanfield09 commented on src/routes/AdminRoutes.php:
```diff
@@ -596,42 +625,71 @@ function fetch_val($key, $array, $default = 'undefined') {
+        $items_set = new \Ds\Set($items);
 
         if($state != 'undefined' && $oldName != 'undefined'){
-            $mysqli->query("UPDATE Reuse_Locations SET state_id = '$state' WHERE name = '$oldName'");
```
Shame. 🔔  Shame. 🔔  Shame. 🔔

istanfield09 commented on src/routes/AdminRoutes.php:
```diff
@@ -596,42 +625,71 @@ function fetch_val($key, $array, $default = 'undefined') {
+            }
+        }
+
+        // $app->log->debug('ITEM ID: '.$id);
```
Remove.

istanfield09 commented on src/routes/AdminRoutes.php:
```diff
@@ -596,42 +625,71 @@ function fetch_val($key, $array, $default = 'undefined') {
 
         /* Update Mobile Database */
-        reuse_generateXML();
+        // reuse_generateXML();
```
Uncomment.

japfohl commented on public_html/admin/addBusiness.php:
```diff
@@ -83,6 +84,14 @@
         </div><!-- end formgroup -->
+        <div class="form-group">
+          <label>Accepted Items: </label>
+          <select id="select-items" name="select-items" class = "form-control" multiple="multiple">
```
I'm _really_ not a fan of using the select element for the following reasons:

1. It requires the user to know that they need to hold control / command while selecting the items.
2. One accidental click and you can undo all the selections you've previously done.
3. Its hard to know what you've already selected and easily scan / other related elements to see if you should / want to add anything else.

I think we should checkboxes or something like it.  Alternatively, the task of adding / editing selected items might be better suited to another page instead of trying to wrap it into this one.

japfohl commented on public_html/admin/searchBusiness.php:
```diff
@@ -127,9 +127,15 @@
+
+  <label for="accepted_items">Accepted Items: </label>
+  <div class="form-group">
+      <select id="select-items" name="select-items" class="form-control" multiple="multiple">
```
See comment on addbusiness.php

japfohl commented on public_html/js/searchBusFunct.js:
```diff
@@ -9,8 +9,24 @@ var webURL = "http://localhost/Corvallis-Sustainability-ReUse/public_html/index.
 var webURL = "";
 /*This happens when the user presses save in the edit view*/
 
+function getSelectValues(select) {
```
This looks like a duplicate of the same function in BusFunct.js.  Any way we could keeping a little more DRY?

japfohl commented on src/routes/AdminRoutes.php:
```diff
@@ -59,6 +59,35 @@ function fetch_val($key, $array, $default = 'undefined') {
+     * @apiSuccess {string[]} items Returns the IDs of items accepted by that business
+     * @apiSuccess {integer} id ID of that item
+     */
+    $app->get('/business/items/:business_id', function($business_id) {
```
This is a bit nit-picky seeing as we're going to be changing all of this up sooner than later but shouldn't this be:
`GET /business/:id/items`?

japfohl commented on src/routes/AdminRoutes.php:
```diff
@@ -596,42 +625,71 @@ function fetch_val($key, $array, $default = 'undefined') {
+        while($row = $locations_items_relations->fetch_array()){
+            $items_relations[] = $row[0];
+        }
+        $app->log->debug($items_relations);
```
I think you need to activate logging for this to do anything: http://docs.slimframework.com/logging/activate/

japfohl commented on src/routes/AdminRoutes.php:
```diff
@@ -669,6 +727,7 @@ function fetch_val($key, $array, $default = 'undefined') {
         $latitude = $mysqli->real_escape_string(fetch_val('latitude', $_POST, null));
         $longitude = $mysqli->real_escape_string(fetch_val('longitude', $_POST, null));
         $website = $mysqli->real_escape_string(fetch_val('website', $_POST, null));
+        $accepted_items = $mysqli->real_escape_string(fetch_val('items', $_POST, null));
```
Git may take care of this automagically but when you rebase, you may have to change this to `Util::fetch_Val()`

istanfield09 replied:
Did you want this merged into `master` or `rendering-candidate`?

japfohl replied:
Oh dur...! Lol master.  Forgot what we were working on.

istanfield09 commented on composer.json:
```diff
@@ -1,21 +1,22 @@
 {
```
I am (pretty sure) that this is how composer autoformats this file, but I could be wrong.

istanfield09 commented on composer.json:
```diff
@@ -1,21 +1,22 @@
 {
```
I am (pretty sure) this is how composer autoformats this file, but I could be wrong.

japfohl replied:
Did you add it via the composer cli or manually add an entry?  I updated the whole composer file by hand so I may have done it wrong.

istanfield09 commented:
I updated the file via CLI and it inserted that line slightly askew. Not
sure which is incorrect at this point.

On Fri, Nov 10, 2017 at 10:32 AM Joe Pfohl <notifications@github.com> wrote:

> *@japfohl* commented on this pull request.
> ------------------------------
>
> In composer.json
> <https://github.com/Corvallis-Sustainability-Coalition/ReUse-Websites/pull/29#discussion_r150306847>
> :
>
> > @@ -1,21 +1,22 @@
>  {
>
> Did you add it via the composer cli or manually add an entry? I updated
> the whole composer file by hand so I may have done it wrong.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Corvallis-Sustainability-Coalition/ReUse-Websites/pull/29#discussion_r150306847>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACzI3l8qUMzrH-zX7H8yKo1OpmuIEeP5ks5s1Ja5gaJpZM4QUPEu>
> .
>

japfohl commented on public_html/admin/searchBusiness.php:
```diff
@@ -26,6 +26,7 @@
   <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">
   <link href='https://fonts.googleapis.com/css?family=Rubik:700' rel='stylesheet' type='text/css'>
   <script src="../js/jquery-1.11.1.min.js"></script>
+  <script src="../js/businessHelper.js"></script>
```
I dislike the fact that we're generating another HTTP request, _but_... this is an acceptable stop-gap.  I suspect we'll resolve this once the PHP takes care of most of this.

japfohl commented on public_html/js/businessHelper.js:
```diff
@@ -0,0 +1,7 @@
+var getChecked = function () {
```
why declare getChecked as a variable and not a function (i know i know... not a huge difference in JS)?  Wouldn't it make more sense to do:
```javascript
function getChecked() {
     // do stuff
}
```
?
